config/tests/src/test/java/org/jboss/windup/config/TestInsiderRuleProvider.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.windup.config;
+
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.graph.GraphContext;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ */
+public abstract class TestInsiderRuleProvider extends WindupRuleProvider implements Insider {
```

Comment:
In general I agree that the concept can be lifted to SimpleActionRuleProvider.

This would be an interface whose implementation will simplify implementing a simple rule with one condition and one (abstract) operation, correct?